### ISSUE TITLE
About how to calculate AED and APD

### ISSUE THREAD

**Issue by ShiQiBiHua0114** [xg-chu/GAGAvatar#36]
How are the evaluation indicators AED and APD calculated in your paper, and are they obtained by using the Deep3DFaceRecon model to extract the exp and pos parameters and then calculate the Euclidean distance?

**Comment by ZhenglinZhou** [comment]
Same question. A possible answer I found in [OTAvatar](https://github.com/theEricMa/OTAvatar/issues/27#issuecomment-1801217498):

>The APD and AED are the L1 similarity between the pose and expression coefficients extracted from gt and pred portraints using[ DeepFaceRecon](https://github.com/microsoft/Deep3DFaceReconstruction).

**Comment by Time-Lord12th** [comment]
@xg-chu Should AED and APD be averaged over the dimensions? I'm not sure if I have correctly used the results of DeepFaceRecon.

**Comment by xg-chu** [comment]
AED and APD are calculated using these codes, they should be the sum of the outputs and then averaged for the batch dim. (If I remember correctly)
```
aed = torch.nn.functional.l1_loss(pred_eres['exp'], target_eres['exp'])
apd = torch.nn.functional.l1_loss(pred_eres['pose'][:3], target_eres['pose'][:3])
```

**Comment by Time-Lord12th** [comment]
> AED and APD are calculated using these codes, they should be the sum of the outputs and then averaged for the batch dim. (If I remember correctly)
> 
> ```
> all_aeds.append(torch.nn.functional.l1_loss(pred_eres['exp'], target_eres['exp'])) # exp: 1d
> all_apds.append(torch.nn.functional.l1_loss(pred_eres['pose'][:3], target_eres['pose'][:3])) # pose: 1d
> 
> aed = np.mean(all_aeds)
> ...
> ```
Thanks for your quick response！
